client/src/components/History/Content/Dataset/DatasetActions.vue

```diff
@@ -1,152 +1,196 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import {
+    faBug,
+    faChartBar,
+    faInfoCircle,
+    faLink,
+    faQuestion,
+    faRedo,
+    faSitemap,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BButton } from "bootstrap-vue";
+import { computed } from "vue";
+import { useRouter } from "vue-router/composables";
+
+import { type MetadataFiles } from "@/api";
+import { copy as sendToClipboard } from "@/utils/clipboard";
+import localize from "@/utils/localization";
+import { absPath, prependPath } from "@/utils/redirect";
+
+import { type ItemUrls } from ".";
+
+import DatasetDownload from "@/components/History/Content/Dataset/DatasetDownload.vue";
+
+library.add(faBug, faChartBar, faInfoCircle, faLink, faQuestion, faRedo, faSitemap);
+
+interface Props {
+    item: {
+        id: string;
+        extension: string;
+        state: string;
+        creating_job: string;
+        rerunnable: boolean;
+        purged: boolean;
+        name: string;
+        meta_files: MetadataFiles;
+    };
```

Comment:
This is an interesting issue. It doesn't seem to be defined under `model.Dataset.states`. It appears to be a client-only state with an old TODO comment located here: https://github.com/itisAliRH/galaxy/blob/4c686f9a4b3c42b87f5829b4263276b03c379764/client/src/mvc/dataset/states.js#L28.

The comment mentions that this state is in the database. However, I could not find a state assignment similar to `hda.state = noPermission` in the entire code base. So, it seems like this code will never be reached and can potentially be removed. Maybe worth asking in the chat in case someone recalls about this.

Reply:
Yes, that was definitely never a state on the backend. There is however an optional `accessible` property that probably matches what you need ?

Reply:
Thanks, I replaced it with `accessible` here:
https://github.com/galaxyproject/galaxy/pull/17353/commits/ff8c7dc3d790d0212c981c90c390a4af8394483f